data/api/client-server/registration_tokens.yaml

```
@@ -0,0 +1,80 @@
+# Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+swagger: '2.0'
+info:
+  title: "Matrix Client-Server Registration Token API"
+  version: "1.0.0"
+host: localhost:8008
+schemes:
+  - https
+  - http
+basePath: /_matrix/client/v1
+consumes:
+  - application/json
+produces:
+  - application/json
+paths:
+  "/register/m.login.registration_token/validity":
+    get:
+      x-addedInMatrixVersion: "1.2"
+      summary: Query if a given registration token is still valid.
+      description: |-
+        Queries the server to determine if a given registration token is still
+        valid at the time of request. This is a point-in-time check where the
+        token might still expire by the time it is used.
+
+        Servers should be sure to rate limit this endpoint to avoid dictionary
```

```suggestion
        Servers should be sure to rate limit this endpoint to avoid brute force
```

Probably should have caught this in the MSC stage, but dictionary attacks are a more specific type of attack, and "brute force" is a more general term.